src/main/java/com/mongodb/hibernate/jdbc/MongoStatement.java

```diff
@@ -56,7 +60,43 @@ class MongoStatement implements StatementAdapter {
     @Override
     public ResultSet executeQuery(String mql) throws SQLException {
         checkClosed();
-        throw new FeatureNotSupportedException("TODO-HIBERNATE-21 https://jira.mongodb.org/browse/HIBERNATE-21");
+        var command = parse(mql);
+        return executeQueryCommand(command);
+    }
+
+    ResultSet executeQueryCommand(BsonDocument command) throws SQLException {
+        startTransactionIfNeeded();
+        try {
+            var collectionName = command.getString("aggregate").getValue();
+            var collection = mongoClient.getDatabase(DATABASE).getCollection(collectionName, BsonDocument.class);
+            var pipeline = command.getArray("pipeline").stream()
+                    .map(BsonValue::asDocument)
+                    .toList();
+            var cursor = collection.aggregate(clientSession, pipeline).cursor();
+            var fieldNames = getFieldNamesFromProjectDocument(
+                    pipeline.get(pipeline.size() - 1).asDocument().getDocument("$project"));
+            return new MongoResultSet(cursor, fieldNames);
+        } catch (RuntimeException e) {
+            throw new SQLException("Failed to run aggregate command: " + e.getMessage(), e);
```

Comment:
It seems this `try` should be only around the .aggregate command, if this is the text?

Reply:
I changed it to more appropriate
```
throw new SQLException("Failed to execute query: " + e.getMessage(), e);
```
to maintain excepion message consistency with other JDBC message pattern. Is that fine as well?